locations/models.py

```diff
@@ -178,7 +186,8 @@ def get_context(self, request):
 
         # Filter by tag
         tag = request.GET.get('tag')
-        locationpages = LocationPage.objects.filter(tags__name=tag).live()
+        locationsindexset= LocationsIndexPage.objects.live().parent_of(self)
+        locationpages = LocationPage.objects.filter(tags__name=tag).live().descendant_of(locationsindexset[0])
```

Comment:
this could probably be: `locationpages = LocationPage.objects.filter(tags__name=tag).live().descendant_of(self.get_parent())`